lib/signer.ex

```diff
@@ -40,78 +40,85 @@ defmodule AntikytheraAws.Signer do
     - `params` must be decoded query strings, in a list of tuple-2
         - To convert from `Antikythera.Http.QueryParams` (which is `Croma.SubtypeOfMap`), just use `Map.to_list/1`
 
-    Note: "content-type" and "x-amz-content-sha256" headers are required for S3, but not generally.
+    Note: "content-type" header is required for S3, but not generally.
     """
     defun prepare_headers(%Creds{access_key_id: aki, secret_access_key: sak, security_token: st} = _creds,
                           region  :: v[String.t],
                           service :: v[String.t],
                           method  :: v[Method.t],
                           path    :: v[UPath.t],
-                          payload :: v[String.t],
+                          payload :: v[binary],
                           headers :: v[Headers.t],
                           params  :: [{String.t, String.t}]) :: Headers.t do
       if String.contains?(path, "?"), do: raise("`path` must not include query string part!")
+      payload_sha256                          = hex_sha256(payload)
+      {sign_ready_headers, amz_date}          = canonicalize_headers_with_amz_date(headers, st, service, payload_sha256)
+      {canonical_request, signed_headers_str} = make_canonical_request_with_shs(sign_ready_headers, method, path, payload_sha256, params)
+      signature                               = compute_signature(aki, sak, amz_date, region, service, canonical_request, signed_headers_str)
+
```

Comment:
In general empty lines in function bodies indicate that the function should be split into multiple functions.
Remove this empty line (or define function for the separated parts).

Reply:
fixed in 3c31fd3